drivers/tty/serdev/pibridge.c

```diff
@@ -260,8 +270,11 @@ int pibridge_req_gate_tmt(u8 dst, u16 cmd, u8 *snd_buf, u8 snd_len,
 	if (dst == PIBRIDGE_BC_ADDR)
 		return 0;
 
-	if (pibridge_recv_timeout((u8 *) &pkthdr, sizeof(pkthdr), tmt) !=
-	    sizeof(pkthdr)) {
+	ret = pibridge_recv_timeout((u8 *) &pkthdr, sizeof(pkthdr), tmt);
+	if (ret == -ERESTARTSYS)
+		return ret;
+
+	if (ret) {
```

Comment:
Independently of the problem you seek to solve, the `ret = ...; if (ret) ...` pattern is IMHO easier to read and closer to kernel coding style than having a function in an if-condition which has significant side effects.